apps/src/codebridge/hooks/useInitialSources.ts

```diff
@@ -5,22 +5,33 @@ import {getAppOptionsEditBlocks} from '@cdo/apps/lab2/projects/utils';
 import {ProjectSources} from '@cdo/apps/lab2/types';
 import {useAppSelector} from '@cdo/apps/util/reduxHooks';
 
+// Hook for getting the initial sources for the current level.
+// If we are in start mode, we show the start sources or the default sources
+// if the level has no start sources.
+// If the user has a project, we prioritize that over the start sources. If there is
+// no project or start sources, we use the default sources.
 export const useInitialSources = (defaultSources: ProjectSources) => {
-  const labState = useAppSelector(state => state.lab);
+  const labInitialSources = useAppSelector(state => state.lab.initialSources);
```

Comment:
I split out the values we needed here because `labState` changes on every project save, which was causing `useMemo` to fire too often.